server/src/main/java/org/elasticsearch/action/support/PlainActionFuture.java

```diff
@@ -413,7 +413,10 @@ private boolean assertCompleteAllowed() {
                 + " with waiter on thread "
                 + waiter
                 + ", could deadlock if pool was full\n"
-                + ExceptionsHelper.formatStackTrace(waiter.getStackTrace());
+                + "waiter stacktrace\n"
+                + ExceptionsHelper.formatStackTrace(waiter.getStackTrace())
+                + "\ncompleter stacktrace\n"
+                + ExceptionsHelper.formatStackTrace(Thread.currentThread().getStackTrace());
```

Comment:
This should be output by the assertion's stack trace, it was omitted on purpose. Is there a case where we are not dumping this?

Reply:
It *is* included in the specific case [here](https://gradle-enterprise.elastic.co/s/mjgqjz3uki7kk/tests/task/:x-pack:plugin:ccr:internalClusterTest/details/org.elasticsearch.xpack.ccr.LocalIndexFollowingIT/testRemoveRemoteConnection?focused-execution=1&top-execution=1#L103)

I suggest to revert this part.

Reply:
You are right. I reverted it in b11a09ac957c88a1bc4fd193278c2ca8be7ab7a6